http.go

```diff
@@ -293,6 +297,40 @@ func (resp *Response) BodyWriter() io.Writer {
 	return &resp.w
 }
 
+// BodyStream returns io.Reader
+//
+// You must close it after you use it.
+func (resp *Response) BodyStream() io.ReadCloser {
+	if resp.bodyStream == nil {
+		resp.bodyStream = bytes.NewReader(resp.Body())
+	}
+	return newCloseReader(resp.bodyStream, resp.closeBodyStream)
+}
+
+type closeReader struct {
+	io.Reader
+	closeFunc func() error
+	closeOnce sync.Once
+	err       error
+}
+
+func newCloseReader(r io.Reader, closeFunc func() error) io.ReadCloser {
+	if r == nil {
+		panic(`BUG: reader is nil`)
+	}
+	return &closeReader{Reader: r, closeFunc: closeFunc}
+}
+
+func (c *closeReader) Close() error {
+	if c.closeFunc == nil {
+		return nil
+	}
+	c.closeOnce.Do(func() {
+		c.err = c.closeFunc()
+	})
```

Comment:
Why is this `sync.Once` needed here? If `Close` can be called multiple times, shouldn't `closeFunc` then also not be able to be called multiple times?

Reply:
if SetBodyStream is true, the clientConn does not immediately close.  need to close  it manually.  close clientConn does not need to be executed multiple times.

Reply:
For example, `sync.Pool` has the same object and `clientConn` is closed multiple times

Reply:
I'm not sure I understand. Can you show how `Close` would be called multiple times?

Reply:
eg: `stream.Close()` Execute multiple times

```go
resp := AcquireResponse()
resp.StreamBody = true
request := AcquireRequest()
request.SetRequestURI(server.URL + "?chunked=true")
if err := Do(request, resp); err != nil {
	t.Fatal(err)
}
stream := resp.BodyStream()
defer func() {
	// exec multiple times
	stream.Close()
	stream.Close()
	stream.Close()
}()
content, _ := io.ReadAll(stream)
if string(content) != "0123456789" {
	t.Fatalf("unexpected body content, got: %#v, want: %#v", string(content), "0123456789")
}
```

Reply:
I thought it wouldn't be good to return an io.ReadCloser directly, so I added the CloseBodyStream method @erikdubbelboer

Reply:
I did some investigating into this issue, https://github.com/valyala/fasthttp/issues/1504, and it seems it is cause by Close being called multiple times. So I understand the `sync.Once` now. But I'm wondering if we should fix the double close instead.

Reply:
`perIPConn#Close()` method is also wrapped in `sync.Once` ?  It also seems reasonable and easier to deal with

Reply:
I think it depends on whether there is concurrent execution of 'closeBodyStream'. If there is, the same problem will indeed occur, but this is an unreasonable use behavior and concurrency framework defect.  @erikdubbelboer